### DISCUSSION TITLE
app.py No such file or Directory Mac M1

### DISCUSSION THREAD

**Discussion by thatjimupnorth** [lkwq007/stablediffusion-infinity#142] [Q&A]
I followed the install instructions exactly, and the line:

conda activate sd-inf

Works perfectly, but the final line:

python app.py

Says "python: can't open file '/Users/jim/app.py': [Errno 2] No such file or directory

I'm a bit new to all of this. Would really appreciate help. Thanks.

**Comment by foxdiefox** [comment]
having same issues with windows

**Reply by thatjimupnorth** [reply]
OK, so is it giving exactly the same error? Seems like the install process is the same for both Mac and Windows because it uses the cross platform Conda framework. Are there any reports of it actually working as normal? I've got a fresh clean install of macOS and SD itself is working perfectly.

**Comment by thatjimupnorth** [comment]
I'm going to answer my own question, since the always willing to help community basically ignored this for weeks and I'm still waiting for someone to reply to four different requests in the Discord Tech Support.  

The install instructions for Windows and Mac are wrong - at least that's to say they're worded very poorly.  

The actual correct process for installing is as follows:

git clone --recurse-submodules https://github.com/lkwq007/stablediffusion-infinity
cd stablediffusion-infinity
conda create -n sd-inf python=3.10
conda activate sd-inf
conda install pytorch torchvision torchaudio -c pytorch-nightly
conda install scipy scikit-image
conda install -c conda-forge diffusers transformers ftfy accelerate
pip install opencv-python
pip install -U gradio
pip install pytorch-lightning==1.7.7 einops==0.4.1 omegaconf==2.2.3
pip install timm

You should then be able to...

conda activate sd-inf
python app.py

...and so long as you're in the (sd-inf) your@name-here stablediffusion-infinity % directory python app.py will run, show you the web address to point your browser at, then when the Gradio interface opens, enter your HuggingFace Token before you click Start.

Hope that helps.
